### PR TITLE
Fixing https://wso2.org/jira/browse/APPCLOUD-306

### PR DESCRIPTION
**Pull request by kasundsilva**

### CODE REVIEW

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag [hunk 1]
```diff
@@ -28,65 +28,85 @@ var server = carbon.server;
...
-	if (userName.indexOf("@") != -1) {
-		tenantDomain = userName.substring(userName.lastIndexOf('@') + 1);
-	} else {	// else assume super tenant
-		tenantDomain = SUPER_TENANT_DOMAIN_NAME;
-	}
-	var ws = require('ws');
-	var context=multitenancy.getPrivilegedCarbonContext();
- 	var tenantManager= multitenancy.getTenantManager();
-	var finalResult={};
+    log.debug("Initializing login module");
```
Comment (danushkaf) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: dont we need if debug enabled

Comment (mcvidanagama) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: add isDebugEnabled check

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag [hunk 2]
```diff
@@ -28,65 +28,85 @@ var server = carbon.server;
...
-	    throw e;
+    // check whether given tenant domain is valid
+    try {
+       log.debug("Checking whether given tenant:" + tenantDomain + " is valid");
+       if (tenantManager.getTenantId(tenantDomain) == -1) {
+           finalResult.error = "true";
+           finalResult.message = "Authorization Failed.";
+           finalResult.username = userName;
+           log.debug("Failing authorization since given tenant:" + tenantDomain + " is not valid");
+        return stringify(finalResult);
```
Comment (mcvidanagama) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: format

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag [hunk 3]
```diff
@@ -28,65 +28,85 @@ var server = carbon.server;
...
         context.startTenantFlow();
         context.getThreadLocalCarbonContext().setTenantId(tenantManager.getTenantId(tenantDomain));
         context.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
         context.getThreadLocalCarbonContext().setUsername(userName);
-		var result = server.Server.prototype.authenticate(userName,password);
-		if(result === true){
-		    var permission = modManager.getPropertyValue(jagg.findSelectedCloud(request.getRequestURL()) + LOGIN_PERMISSION);
+        var endTime_tenantFlow = new Date().getTime();
+        log.debug("Tenant flow started in :" + (endTime_tenantFlow-startTime_tenantFlow) + " milliseconds");
+        log. debug("Start authenticating user:" + userName + " in tenant:" + tenantDomain);
```
Comment (mcvidanagama) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: log.debug has space between "."

File: modules/jaggeryapps/appmgt/src/site/blocks/user/login/ajax/login.jag [hunk 4]
```diff
@@ -25,43 +25,51 @@ var log=new Log();
...
+                finalResult.message = "Please provide userName & password";
+                print(finalResult);
+                return;
+            }
+            try {
+                log.debug("Calling login module to authenticate");
+                var startTime_authcheck = new Date().getTime();
+                result = mod.login(userName, password, requestURL);
+                var parsedJSON = eval('('+result+')');
+                var errorStatus=parsedJSON.error;
```
Comment (mcvidanagama) [inline modules/jaggeryapps/appmgt/src/site/blocks/user/login/ajax/login.jag]: format

**Pull request merged**